src/JitInterface/src/CorInfoTypes.cs

```diff
@@ -1241,7 +1253,12 @@ public struct VarLoc
        public int vlType;
        // The 64bit field is here to keep VarLoc 8byte aligned on Amd64.
        // For x86, we need to change the VarLoc definition here.
-       public long A;   
+#if BIT64
```

Comment:
That's weird. The nice way to suppress these warnings is to mark the struct as Sequential:

```
[StructLayout(LayoutKind.Sequential)]
struct VarLoc
{
    IntPtr A; // vlType + padding
    int B;
    int C;
    int D;
}
```